tests/test_track_check_qc_functions.py

```diff
@@ -143,9 +145,63 @@ def test_do_track_check_mixed(ship_frame):
     )
     for i in range(len(trk)):
         if i == 15:
-            assert trk[i] == 1
+            assert trk[i] == failed
         else:
-            assert trk[i] == 0
+            assert trk[i] == passed
+
+
+def test_do_track_check_testdata():
+    vsi = [np.nan] * 10
+    dsi = [np.nan] * 10
+    lat = [46.53, 46.31, 46.09, 45.87, 45.88, 46.53, 46.31, 46.09, 45.87, 45.88]
+    lon = [
+        -13.17,
+        -12.99,
+        -12.81,
+        -12.62,
+        -12.57,
+        -13.17,
+        -12.99,
+        -12.81,
+        -12.62,
+        -12.57,
+    ]
+    date = np.array(
+        [
+            "1873-01-01T01:00:00.000000000",
+            "1873-01-01T05:00:00.000000000",
+            "1873-01-01T09:00:00.000000000",
+            "1873-01-01T13:00:00.000000000",
+            "1873-01-01T17:00:00.000000000",
+            "1875-01-01T01:00:00.000000000",
+            "1875-01-01T05:00:00.000000000",
+            "1875-01-01T09:00:00.000000000",
+            "1875-01-01T13:00:00.000000000",
+            "1875-01-01T17:00:00.000000000",
+        ]
+    )
+    date = pd.to_datetime(date).tolist()
+
+    results = do_track_check(
+        vsi=vsi,
+        dsi=dsi,
+        lat=lat,
+        lon=lon,
+        date=date,
+    )
+    expected = [
+        passed,
+        passed,
+        failed,
+        passed,
+        passed,
+        passed,
+        passed,
+        failed,
+        passed,
+        passed,
+    ]
+    np.testing.assert_array_equal(results, expected)
```

Comment:
@jjk-code-otter: I extracted these inputs from our test dataset. Before my changes, the track check returned expected QC flags. Now thr QC flags are always ``passed``. I have no idea why. The unittests all run like expected. Could you figure out, what QC flags to expect for this test?